scripts/release_consumer_smoke_aggregate.py

```
@@ -0,0 +1,136 @@
+    release_kind: str,
+    allowed_not_supported: set[str] | None = None,
+) -> dict[str, Any]:
+    entries: dict[str, dict[str, Any]] = {}
+    for index, entry in enumerate(receipts):
+        kind = entry.get("kind")
+        key = entry.get("artifact") or kind
+        if kind == "binary":
+            key = entry.get("surface")
+        if not key:
+            key = f"invalid-receipt-{index}"
+            entry = {
```

Ruff (0.16.0)

[warning] `for` loop variable `entry` overwritten by assignment target

(PLW2901)